README (#5)
zywangdylan/lewagon-mp-dashboard: README.md

# LeWagon WeChat Mini Program Dashboard

This is an all-in-one dashboard build with [Vue3](https://vuejs.org/) + [Ant Design Vue](https://antdv.com/docs/vue/introduce) to manage all the main LeWagon WeChat Mini Programs(WXMP), currently Course WXMP and Event WXMP, so as to provide more conveniences to LeWagon team for manipulating all WXMPs.

## Getting Started
---
**Frontend**

> 1. `git clone` this repo to your local

Reviewer: ```suggestion
1. `git clone` this repo
```

2. Ask Slack `#china-engineering` group to add you as a collaborator on this github repo.

**Backend**
1. Setup your [Minapp](https://cloud.minapp.com/) account. You can refer to the guide [here](https://www.notion.so/lewagon/minapp-Documentation-e8e0cab4a5844658a5132f7ac5f548aa). Please make sure you can access both `LeWagonCourses` and `LeWagonEvents`.

## Architecture
---

> You can consider the dashboard as another frontend, same as Course/Event WXMP, that is connected with the minapp and able to manipulate some functions and data.

Reviewer: You can consider the dashboard as another frontend, same as Course/Event WXMP, that is connected with the minapp and able to **manipulate some functions and data**.

Please specify what do you mean. Can this dashboard actually *change* (manipulate) functions on MinApp? Or it is just to do CRUD operations on courses and events? Or more than that?

```bash
npm run serve
```

**Compiles and minifies for production**
```bash
npm run build
```

### Vue 3
This project is using [Vue 3](https://vuejs.org/guide/introduction.html), a JavaScript framework for building user interfaces, with the support of some other libraries under the Vue system:

- `Vue ~> 3.0.0` 
- `vue-request ~> 1.2.4` 
    Vue composition API for data fetching, supports SWR, polling, error retry, cache request, pagination, etc. 

    [Github](https://github.com/attojs/vue-request) | [Doc](https://www.attojs.org/)

- `vue-router ~> 4.0.12`
    Vue Router is the **official router** for Vue.js. It deeply integrates with Vue.js core to make building Single Page Applications with Vue.js a breeze. 

    [Github](https://github.com/vuejs/router) | [Doc](https://router.vuejs.org/installation.html)

- `vuex ~> 4.0.2`
    Vuex is a **state management pattern + library** for Vue.js applications. It serves as a centralized store for all the components in an application, with rules ensuring that the state can only be mutated in a predictable fashion.

    [Github](https://github.com/vuejs/vuex) | [Doc](https://vuex.vuejs.org/)

### Ant Design Vue

> [Ant Design Vue](https://antdv.com/components/overview/) provides plenty of UI components to enrich the web applications including typography, layouts, tables etc. By using Ant Design Vue in the dashboard, we can a consistent design system across all the pages. 🪄

Reviewer: ```suggestion
[Ant Design Vue](https://antdv.com/components/overview/) provides plenty of UI components to enrich web applications including typography, layouts, tables etc. By using Ant Design Vue in the dashboard, we can have a consistent design system across all the pages. 🪄
```

Before developing a new feature with a new design, please have a look at the components in Ant Design Vue and you can start from that which can save you lots of time and work.

1. Add `import` and `.use()` in `src/main.js` so that you can use the component anywhere in your app.

```js
import { Avatar } from 'ant-design-vue';
app.use(Avatar)
```
2. Use the corresponding tag in the `.vue` file.

```vue
<template>
    <a-avatar 
      :size="64"
      :style="{ backgroundColor: red, verticalAlign: 'middle' }"
    >
        Dy
    </a-avatar>
</template>
// ...
```

**sass**

> `sass` is used in some `.vue` file for conveniences such as changing some styles in Ant Design components, using `sass` syntax etc.

Reviewer: ```suggestion
`sass` is used in some `.vue` file for conveniences such as changing some styles in Ant Design components with following libraries:
```

```json
"node-sass": "^4.14.1",
"sass-loader": "^10.2.1"
```

### `src` Structure
Since the dashboard is controlling multiple MPs, the files are also separated into different folders named in the MP names in the main folders. 

For example, in `components`, we have `.vue` files in `Courses` and `Events` which represent the components used in `Courses` and `Events` pages correspondingly.

> ![Events & Courses Tabs](https://cloud-minapp-37096.cloud.ifanrusercontent.com/1oE4tUATY2LUzETW.png)

Reviewer: Let's add a full-page screenshot of both tabs side-by-side instead

> ### How to connect with backend (minapp BaaS)?

Reviewer: ```suggestion
### Interacting with the backend (minapp BaaS)
```
